### ISSUE TITLE
[bug] Backtest engine reports an empty data load as a 0-trade success and annualizes Sharpe incorrectly for intraday windows

### ISSUE THREAD

**Issue by davdunc** [TradingAsBuddies/falcon-core#21]
## Observed (live, /api/backtests/summary)

```
vwap_bounce            runs=100 trades=0     ret=0        sharpe=0
market_memory          runs=60  trades=0     ret=0        sharpe=0
one_candle_rule        runs=3   trades=0     ret=0        sharpe=0
volatility_squeeze     runs=4   trades=0     ret=0        sharpe=0
atr_breakout           runs=100 trades=1914  ret=-7.4%    sharpe=-6.97
opening_range_breakout runs=68  trades=207   ret=-0.77%   sharpe=-1.12
```

Four strategies have never traded in hundreds of runs; atr_breakout's Sharpe is not a number anyone can act on.

## Root causes

1. `src/falcon_core/backtesting/engine.py:302-314` returns a fully-zeroed `BacktestResult` when `trades.empty`, with **no distinction between "no signal" and "data load failed"**. With `polygon_client.py:155`'s former `!= "OK"` gate (SPEC.md's "enabling bug"), 100 runs / 0 trades is most likely an empty feed, not a strategy declining to trade. `market_memory` / `volatility_squeeze` (5m) and `one_candle_rule` (1m) share that path. (#11 covers the seeded strategies having no working code — this issue is about the engine hiding it.)
2. Sharpe: `engine.py:320-324` annualizes with `years = max(days/365.25, 0.01)`, so a short intraday window raises total return to the ~100th power; `engine.py:346-350` then divides by volatility from a **trade-indexed** equity curve scaled by `sqrt(252)`. 1914 trades at 20 bps round-trip (`engine.py:223, 235`) guarantees a large negative number regardless of edge.
3. `BTBacktestEngine` (`engine.py:416-470`): if `bt`/`ffn` are missing, `_check_dependencies` (`:431-441`) only warns and `run` silently falls back to `SimpleBacktestEngine` (`:449-454`). No minimum-bar guard, so the `backtest-engine` sentinel (5 bars, 0 trades) always passes and tests nothing.

## Proposed fix

- Zero bars → `BacktestResult(status='error', reason='no_data')`; surface `bars_loaded` on every run row.
- Sharpe from a time-indexed equity curve at the bar frequency (`sqrt(bars_per_year)`), plus Deflated/Probabilistic Sharpe for multi-run summaries.
- `raise` when `len(data) < strategy.min_bars`; sentinel runs a golden fixture (~60 bars) and asserts `> 0` trades.
- Record the engine actually used on the run row.

**Comment by davdunc** [comment]
Partially addressed by #23.

**Fixed there:** the Sharpe computation (root cause 2) — daily time-indexed equity curve, standard mean/std form, plus `metrics_reliable` so the promotion gates refuse to act on a ratio computed from too few observations. Also the `gross_loss = 0.01` default, which turned an undefined profit factor into `gross_profit × 100`.

**Still open — root cause 1:** an empty data load is still reported as a 0-trade success. `engine.py` returns a fully-zeroed `BacktestResult` whether the strategy declined to trade or the feed returned nothing, which is why `vwap_bounce`, `market_memory`, `one_candle_rule` and `volatility_squeeze` sit at 100 runs / 0 trades looking like a strategy problem. I have `status='error', reason='no_data'` plus `bars_loaded` on every run row, to rebase onto #23 once it merges.

@minibar — do those four strategies load a non-zero bar count on the deployed stack? That is what separates this from #11 (empty shells), and you are the one who can answer it.

**Comment by davdunc** [comment]
PR #24 landed the `no_data` error result (`engine.py:255-275`, `_error_result` :538-563), BT-fallback logging with `engine_used` (:600-609), and no sub-session annualisation (:410-417). Three things keep this open:

1. **Sharpe is on the wrong time base, now scaled harder.** `equity` is still trade-indexed (`_build_equity_curve` :497-509) but annualisation uses `_bars_per_year(data)` (:446-453) — on 1-minute bars that is 98,280 periods/yr, so per-trade mean/std is multiplied by ~313 instead of ~16. The old `sqrt(252)` was wrong; this is more wrong for intraday. Fix: mark-to-market equity at bar frequency, or scale per-trade returns by trades-per-year. PR #25's `metrics_reliable` gate (`MIN_RETURN_OBSERVATIONS=20`) and `gross_profit/gross_loss` fields are worth keeping when the two are reconciled (they conflict in this hunk).
2. **`status/reason/bars_loaded/engine_used` are dropped before persistence.** `exec_api._summary_from_result` (:163-175) and `_build_record` (:178-200) don't carry them; `results_api.store_backtest_run` (:208-231) INSERT is unchanged; `scheduler.py:304-316` builds `BacktestRunRecord` without them. A `no_data` run is still stored as a 0-trade, Sharpe-0 row — the dashboard cannot tell the difference. Touches #10 (schema).
3. `min_bars` guard is inert — `getattr(strategy, 'min_bars', 0)` at :260, and no strategy defines it.

No tests cover the engine changes.
